src/hotspot/share/code/compressedStream.hpp

```diff
@@ -111,4 +112,102 @@ class CompressedWriteStream : public CompressedStream {
   }
 };
 
+class CompressedSparseData : public ResourceObj {
+protected:
+  u_char* _buffer;
+  int     _position; // current byte offset
+  size_t  _bit_position {0}; // current bit offset
+
+public:
+  CompressedSparseData(u_char* buffer = NULL, int position = 0) {
+    _buffer   = buffer;
+    _position = position;
+  }
+
+  u_char* buffer() const { return _buffer; }
```

Comment:
Please check my change.
- I move _buffer from CompressedSparseData to CompressedSparseDataReadStream and CompressedSparseDataWriteStream - the first one is constant.
- I update debugInfoRec.cpp, nmethod.cpp, nmethod.hpp. On the one hand, I would like to avoid a lot of file chaining. On the other hand, the const modifier adds readability.
Is that what you meant?